src/main/java/com/google/step/snippet/servlets/SearchServlet.java

```diff
@@ -102,22 +106,39 @@ public void doGet(HttpServletRequest request, HttpServletResponse response)
   }
 
   private String getLink(String id, String query) {
-    String cse_request = CSE_URL + "?key=" + API_KEY + "&cx=" + id + "&q=" + query;
-    HttpRequest linkRequest =
-        HttpRequest.newBuilder()
-            .GET()
-            .uri(URI.create(cse_request))
-            .setHeader("User-Agent", "Java 11 HttpClient Bot")
-            .build();
-    HttpResponse<String> linkResponse;
+    String url = CSE_URL + "?key=" + API_KEY + "&cx=" + id + "&q=" + query;
+    CloseableHttpClient httpClient = HttpClients.createDefault();
+    CloseableHttpResponse response;
     try {
-      linkResponse = httpClient.send(linkRequest, HttpResponse.BodyHandlers.ofString());
-    } catch (IOException | InterruptedException ex) {
+      response = httpClient.execute(new HttpGet(url));
```

Comment:
@anbenson mentioned using try-with-resources idiom to catch exceptions and close resources.

The top answer demonstrates how to do this at the end of their answer: https://stackoverflow.com/questions/21574478/what-is-the-difference-between-closeablehttpclient-and-httpclient-in-apache-http

I think to do this, you'll have to move more of your logic into the try/catch as you'll need to finish consuming the response before closing the resources. Regardless of whether you use the try-with-resources or not, you'll want to close the resources somehow.

I'm less familiar with this so I don't want to lead you astray - @anbenson could you confirm that my comment is correct (or horribly incorrect) with regards to using try-with-resources?

Reply:
I am no expert with try-with-resources, but my understanding is that it's like Python's `with open("file.txt", "r") as f:` syntax, so I suppose so.